abstract/imaging_device.py

```diff
@@ -1,14 +1,85 @@
 # extended from https://gist.github.com/HazenBabcock/8d1042f564dcaee60c3a3972f2d56999
 
 
+class HWProperty(object):
+    """Defines a hardware property.
+    """
+    def __init__(self, name = None, value = None, limits = None, values = None, writable = None, readable = None, **kwds):
+        super().__init(**kwds)
+
```

Comment:
Should include a data type (float/int/bool/str)

Reply:
Agreed. There should also probably be a descriptor field as well.

Reply:
Another possibility would be to have sub-classes of `HWProperty` for each type of property (`HWPropertyFloat`, `HWPropertyInt`,..) . Not sure whether this is a better or worse idea than having a single class with a data type property.

Reply:
Let's write it out and see. If the code gets messy (especially when considering type=array), then I'm not opposed to subclassing.

Reply:
I've added these changes in the most recent commit.